swan-lake/resources/featured-scenarios/create-your-first-connector-with-ballerina.md

```diff
@@ -0,0 +1,170 @@
+   - `types.bal`: Contains the data types used in the client.
+   - `utils.bal`: Contains utility functions used in the client.
+
+## Step 4: Test the connector
+
+Now that your connector is generated, it is important to write tests to ensure everything works as expected.
+
+1. In the `ballerina/tests` directory, add the required test files and write test cases for the key operations of your connector. Aim to cover as many API use cases as possible.
+
+2. Execute the tests using the `bal test` command to verify the functionality of your connector.
+
+> For detailed guidance on writing tests, check the [Ballerina testing guide](https://ballerina.io/learn/test-ballerina-code/test-a-simple-function/).
+
+> For sample implementations, refer to the [test implementation of the Ballerina Twitter connector](https://github.com/ballerina-platform/module-ballerinax-twitter/tree/main/ballerina/tests).
+
+## Step 5: Document the connector
+
+Follow these steps to ensure your connector is well-documented:
+
+1. Update `Module.md` and `Package.md` files
+
+   These files will be displayed on the Ballerina Central package landing page. Ensure they introduce and explain the connector package clearly and comprehensively. Each file should include the following sections:
+   
+   - **Overview**: Provide a concise introduction to the connector, explaining its purpose and key features.
+   - **Setup**: Offer step-by-step instructions on configuring the connector and any necessary prerequisites, such as API keys or environment setup.
+   - **Quickstart**: Include a simple, clear example that enable users to start using the connector immediately.
+   - **Examples**: Link to additional use cases, providing context on how the connector can be used in different scenarios.
+
+   > For reference, check the [Twitter connector documentation](https://github.com/ballerina-platform/module-ballerinax-twitter/blob/main/ballerina/Module.md).
+
+2. Update the `README.md` file
+
+   This file will be displayed on the GitHub repository landing page. Therefore, it should include the same information as `Module.md` with a few additional sections such as `Building from Source`, `Contributing`, `License`, etc.
+   
+   > For reference, check the [Twitter connector README](https://github.com/ballerina-platform/module-ballerinax-twitter/blob/main/README.md).
+
+3. Write example use cases (optional)
+
+   Providing practical examples helps users understand the connector better. These examples should show how the connector is used in real-world scenarios.
+
+   - All the examples should be added under the `examples/` directory.
+   - Each example should be added as a Ballerina package with its own `README.md` file, explaining the use case and how to run the example.
+
+   > For reference, check the [Twitter connector examples](https://github.com/ballerina-platform/module-ballerinax-twitter/tree/main/examples).
+
+## Step 6: Publish the connector
```

Comment:
I assume the connector template has Github workflows (connector related), which includes a connector publishing workflow. Wouldn't it be better if we provide the information to setup the BCentral token in the Github repository so that they could just use the workflow to publish the connector ?

Reply:
Since our standard workflow templates are having other secrets in addition to the central token to be set up, asking them to follow the standard publishing procedure would make this guide simpler IMO. wdyt?